src/Illuminate/Database/Eloquent/Builder.php

```diff
@@ -829,6 +885,32 @@ protected function callScope($scope, $parameters)
     }
 
     /**
+     * Get the underlying query builder instance with applied global scopes.
+     *
+     * @return \Illuminate\Database\Query\Builder|static
+     */
+    public function getQueryWithScopes()
+    {
+        if (! $this->scopes) {
+            return $this->getQuery();
+        }
+
+        $builder = clone $this;
+
+        foreach ($this->scopes as $scope) {
+            if ($scope instanceof Closure) {
+                $scope($builder);
+            }
+
+            if ($scope instanceof ScopeInterface) {
```

Comment:
What about throwing an exception in the Model class? I think it's a good practice to throw an Exception as soon as possible for easier debugging.